### PR TITLE
feat: Create the @deepnote/convert package for converting Jupyter notebooks to Deepnote notebooks.

### PR DESCRIPTION
**Pull request by Artmann**
We'll use this in the VS Code extension to help users migrate from Jupyter notebooks.

## Release Notes

* **New Features**
  * Introduced a new conversion package enabling CLI and programmatic conversion of Jupyter notebooks to Deepnote format.

* **Documentation**
  * Added comprehensive documentation for the conversion package, including installation, CLI usage, and examples.

* **Chores**
  * Updated CI workflow with explicit build steps before typechecking and testing.
  * Enhanced gitignore patterns and spell-check dictionary.

## Summary by CodeRabbit

* **New Features**
  * Added a new convert package with CLI and programmatic APIs to convert Jupyter notebooks into Deepnote projects.

* **Documentation**
  * Added a comprehensive README with installation, CLI usage, options, and examples.

* **Tests**
  * Added extensive test suites and notebook fixtures covering single/multi-file conversion, edge cases, deterministic IDs/timestamps, outputs, and end-to-end behavior.

* **Chores**
  * CI now builds packages before typechecking/tests; updated ignore patterns, spell-check dictionary, package publish config, and coverage exclusions.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -99,6 +99,9 @@ jobs:
       - name: Install dependencies
         run: pnpm install --frozen-lockfile
 
+      - name: Build packages
```
Comment (Artmann) [inline .github/workflows/ci.yml:102]: This is required as we have dependecies between the packages in the mono repo.

**Comment by Artmann** [comment]
@CodeRabbitai Can you review this again with the latest changes?

File: packages/convert/src/cli.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+      if (stat?.isDirectory()) {
+        return resolve(absoluteOutputPath, outputFilename)
+      }
+
+      return absoluteOutputPath
+    }
+
+    return resolve(cwd, outputFilename)
+  }
```
[the remark this answers is not in the document]
Comment (Artmann) [reply packages/convert/src/cli.ts:42]: Fixed

File: packages/convert/src/cli.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  })
+}
+
+// Only run main if not in test mode
+if (process.env.NODE_ENV !== 'test' && process.env.VITEST !== 'true') {
+  main().catch(error => {
+    console.error(error)
+    process.exit(1)
+  })
+}
```
[the remark this answers is not in the document]
Comment (Artmann) [reply packages/convert/src/cli.ts]: Fixed

File: packages/convert/src/jupyter-to-deepnote.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      notebooks: [],
+      settings: {},
+    },
+    version: '1.0.0',
+  }
+
+  for (const filePath of inputFilePaths) {
+    const filename = filePath.split('/').pop()
+    const extension = filename?.split('.').pop()?.toLowerCase().trim()
+    const name = filename ? basename(filename, `.${extension}`) : 'Untitled notebook'
```
[the remark this answers is not in the document]
Comment (Artmann) [reply packages/convert/src/jupyter-to-deepnote.ts]: Fixed

File: packages/convert/src/jupyter-to-deepnote.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    const ipynbJson = await fs.readFile(filePath, 'utf-8')
+    const ipynb = JSON.parse(ipynbJson) as IpynbFile
```
[the remark this answers is not in the document]
Comment (Artmann) [reply packages/convert/src/jupyter-to-deepnote.ts]: Fixed

File: packages/convert/src/jupyter-to-deepnote.test.ts [hunk 6]
```diff
@@ -10,15 +10,25 @@ import { convertIpynbFilesToDeepnoteFile } from './jupyter-to-deepnote'
 vi.mock('uuid', async () => {
   const actual = await vi.importActual<typeof import('uuid')>('uuid')
   let counter = 0
+  const mockV4 = vi.fn(() => {
+    counter++
+    return `test-uuid-${counter.toString().padStart(3, '0')}`
```
Comment (saltenasl) [inline packages/convert/src/jupyter-to-deepnote.test.ts:15]: I wouldn't use counter tbf because if i add a new test at the beginning of the file it will require me to regenerate all of the snapshots after that test because the counter would be incremented in those
